### DISCUSSION TITLE
Avoid redundant template for slot

### DISCUSSION THREAD

**Discussion by ShenQingchuan** [vuejs/rfcs#301] [General]
## How the things be like now ?

For example, when using slots, we put something into a `<template #xxx> </template>`, but there are many cases with single root slot node...

I copied a code snippet from  [ant-design-vue's comment UI component ](https://2x.antdv.com/components/comment), as my example :

```vue
<a-comment>
    <template #actions>
      <span key="comment-nested-reply-to">Reply to</span>
    </template>
    <template #author>
      <a>Han Solo</a>
    </template>
    <template #avatar>
      <a-avatar
        src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
        alt="Han Solo"
      />
    </template>
    <template #content>
      <p>
        We supply a series of design principles, practical patterns and high quality design
        resources (Sketch and Axure).
      </p>
    </template>
</a-comment>
```

## What do we want ?
I was wondering that if we can just extend the directive `v-slot:xxx`(or short as `#xxx`) 
to make it can also be used on a normal element/Vue component rather than just `<template>`

and then we can get our template more readability. It would be a great DX imporving for us...

```vue
<a-comment>
    <span #actions key="comment-nested-reply-to">Reply to</span>
    <a #author>Han Solo</a>
    <a-avatar
      #avatar
      src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
      alt="Han Solo"
    />
    <p #content>
      We supply a series of design principles, practical patterns and high quality design
      resources (Sketch and Axure).
    </p>
</a-comment>
```

**Comment by Justineo** [comment]
You may want to read our very first RFC: [0001-new-slot-syntax.md](https://github.com/vuejs/rfcs/blob/master/active-rfcs/0001-new-slot-syntax.md#motivation).
